.github/workflows/test_ci_cases.yaml

```diff
@@ -0,0 +1,46 @@
+name: CI tests
+
+on: 
+  push:
+    branches: [ "main" ]
+
+  pull_request:
+    branches: [ "main" ]
+
+jobs: 
+  CI-tests:
+    runs-on: self-hosted
+    steps: 
+
+      - name: Checkout latest enclave-cc source code
+        if: ${{ github.event_name == 'push' }}
+        uses: actions/checkout@v3
+        with:
+          ref: main
+
+      - name: Checkout PR source code
+        if: ${{ github.event_name == 'pull_request' }}
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+                
+      - name: Install CoCo operator
+        run: ./ci/ci_tests.sh install_coco_operator
+
+      - name: Install Enclave-CC runtimeclass payload
```

Comment:
one problem to think about is how to get a new payload built for testing purposes. For example, a pull request to enclave-cc repository is created that changes agent-enclave.  if we deploy "quay.io/confidential-containers/runtime-payload-ci:enclave-cc-HW-latest", it does not contain the changes we want to test.

Reply:
Let the yaml file which will push the payload image to quay.io waits for security scanning by quay.io until it succeeds. And set this ci test yaml file depends on that yaml file. I think we can open a new issue to discuss this problem in detail. : )

Reply:
we don't publish payloads for PRs but only when new code is merged in `main` so that will not work. for PRs one simple approach would be to just have a matrix job for both HW and SW

Reply:
I don't understand. can you give a simple example scheme? When the code of the agent is changed, how can I use the latest payload?

Reply:
you cannot use the payload images in a CI pipeline because they do not contain the changes you want to test. See `e2e.yaml`: we build the agent-enclave and boot-instance bundlers on the runner and install them manually.

Reply:
Do you mean I can re-use the payload image built by `e2e.yaml`, and run `e2e.yaml` on sgx machine as a HW matrix test?

Reply:
The sgx machine won't be re-created as a fresh new system every time CI tests begin to run. If I set `e2e.yaml` as a matrix test, many installation steps in `e2e.yaml` will be skipped, such as installing crictl, containerd, cni and etc.

Reply:
> Do you mean I can re-use the payload image built by `e2e.yaml`, and run `e2e.yaml` on sgx machine as a HW matrix test?

yes that is what I had in mind but that's just one option. My key message is that for the CI testing PRs we must ensure the code changes in that PR get also tested.

For a robust CI setup, it would be good to always start a fresh setup and not expect the previous round left the system in clean state.

Reply:
Is it possible to use the following solution:  build a docker image in which prerequisites such as occlum runtime, containerd and k8s etc are installed. This image is built nightly and be placed at sgx ci machine. Then every time commits are updated from a PR, the ci machine will start a container from this image, and run build payloads and start a workload etc inside the container.
To use this method, we can run ci tests concurrently if there are a lot of commits at the same time. In addition, we don't need to clean up the environment because every container is fresh new.

Reply:
In the current stage I will add the part of building payload image locally. This is just a simple CI pipeline. And I will implement container-style CI tests at next version. How about this scheme?

Reply:
that works. The "container-style CI" could follow what we use with the operator: the latest payload is tested in a nightly test: https://github.com/confidential-containers/operator/blob/6f241fbc056f0a5d9e1bd2c10b2cedc0782b99ff/.github/workflows/enclave-cc-cicd.yaml#L49-L54

Reply:
The `ImagePullPolicy` should be change to `IfNotPresent` in the following two lines to allow me to test payload images locally:
https://github.com/confidential-containers/operator/blob/18733db5321bf6649a56c7d698f323aecc294800/controllers/ccruntime_controller.go#L878
https://github.com/confidential-containers/operator/blob/18733db5321bf6649a56c7d698f323aecc294800/controllers/ccruntime_controller.go#L686